File: public/Code/LC20/lc20.java
public boolean ans(String s) {
        
    Deque<Character> stack = new LinkedList<>();
    
    for (char c: s.toCharArray()) {
        if (!stack.isEmpty()){
            if ((stack.peekLast() == '(' && c == ')') ||
                (stack.peekLast() == '{' && c == '}') ||
                (stack.peekLast() == '[' && c == ']')){
                stack.removeLast();
                continue;
            }
        }
        stack.addLast(c);
    }
    
    return stack.isEmpty();
}

File: public/Code/LC20/lc20.py
from collections import deque

def ans(s):
    
    stack = deque()
    
    for c in s:
        
        if stack:
            if (stack[-1] == '(' and c == ')') or (stack[-1] == '{' and c == '}') or (stack[-1] == '[' and c == ']'):
                
                stack.pop()
                continue
                
        stack.append(c)
    
    return len(stack) == 0